libgalois/src/PropertyGraph.cpp

```diff
@@ -288,6 +289,77 @@ katana::PropertyGraph::Equals(const PropertyGraph* other) const {
   return true;
 }
 
+std::string
+katana::PropertyGraph::ReportDiff(const PropertyGraph* other) const {
```

Comment:
Maybe some way to return a value to indicate when graphs do match? E.g.,
```cpp
class GraphDiff {
public:
  std::string& message();
  bool is_same();
};

GraphDiff katana::PropertyGraph::Diff(... other) const;
```

Reply:
We already have an `Equals `predicate, so you can test for `Equals` and then call `ReportDiff`.  Is a `GraphDiff `object a better interface for what looks like identical functionality?